admin/broadleaf-open-admin-platform/src/main/resources/open_admin_style/js/admin/components/assetSelector.js

```diff
@@ -170,8 +170,9 @@ $(document).ready(function() {
     $('body').on('click', 'button.edit-asset-selector', function() {
 
         var $modal = BLCAdmin.getModalSkeleton();
-        var primaryData = JSON.parse($("#fields\\'defaultSku__skuMedia---primary\\'\\.value").val());
-        var primaryDatum = primaryData['id'];
+        var hiddenInput = $(this).closest('.asset-selector-container').find(":input:hidden.mediaItem");
+        var parent = $(this).closest('.field-group');
+        var primaryData = JSON.parse(hiddenInput.val());        var primaryDatum = primaryData['id'];
```

Comment:
@agustinlacombadev looks like a line got double up here, let's just move the `var primaryDatum...` to the next line.